src/libraries/System.Console/src/System/IO/KeyParser.cs

```diff
@@ -103,7 +104,7 @@ private static bool TryParseTerminalInputSequence(char[] buffer, TerminalFormatS
         // Is it a four character sequence used by Linux Console or PuTTy configured to emulate it? (examples: '^[[[A' (F1), '^[[[B' (F2))
         if (input[1] == '[' && input[2] == '[' && char.IsBetween(input[3], 'A', 'E'))
         {
-            if (!terminfoDb.TryGetValue(buffer.AsMemory(startIndex, 4), out parsed))
+            if (!terminfoDb.TryGetValue(buffer.AsSpan(startIndex, 4), out parsed))
```

Comment:
Given that we're already using memory-based keys rather than allocating, what are the expected perf benefits from moving to span-based lookup?